### ISSUE TITLE
Cannot style 'create' option of Creatable component

### ISSUE THREAD

**Issue by chrisbutler** [JedWatson/react-select#3377]
I see options to adjust the label, or set the position in the list of options (https://react-select.com/props#creatable-props) but there is seemingly no way to use the `style` prop to change the appearance of only this option

Also, there are no unique or useful classNames added to the create option. I believe this is a regression, and it's certainly a missing feature either way

**Comment by Rall3n** [comment]
@chrisbutler yes, you can style the `create`-option of the select. The styles functions get the props that are being transfered to the components as a second argument. So you get the data that is being set to the `Option` component.  Now you just have to check for `props.data.__isNew__` and style the component according to the value.

```jsx
<Creatable
    { ... }
    styles={{
        option: (providedStyles, props) => ({
            ...providedStyles, // To keep the default style
            color: props.data.__isNew__ && 'red'
        })
    }}
/>
```
[CodeSandbox](https://codesandbox.io/s/lxj31m1j49)
